### ISSUE TITLE
Does not work with Expo 52: Uncaught Error "undefined is not a function"

### ISSUE THREAD

**Issue by ian** [getsentry/sentry-react-native#4341]
### What React Native libraries do you use?

Expo Router, Expo (mobile only)

### Are you using sentry.io or on-premise?

sentry.io (SaS)

### @sentry/react-native SDK Version

6.3.0

### How does your development environment look like?

 (NOBRIDGE) ERROR  TypeError: undefined is not a function [Component Stack]

![Image](https://github.com/user-attachments/assets/4da5e44c-f82c-4b3e-87ad-c14d3442c2d6)

### Sentry.init()

Sentry.init({
  dsn: "...",
})

### Steps to Reproduce

Use "expo": "52.0.14"

Then set your app/_layout.tsx like this:

```
import { Stack } from "expo-router"
import * as SplashScreen from "expo-splash-screen"
import { useEffect } from "react"
import * as Sentry from "@sentry/react-native"

Sentry.init({
  dsn: "...",
})

// Prevent the splash screen from auto-hiding before asset loading is complete.
SplashScreen.preventAutoHideAsync()

function RootLayout() {
  useEffect(() => {
    SplashScreen.hideAsync()
  }, [])

  return (
    <Stack>
      <Stack.Screen name="+not-found" />
    </Stack>
  )
}

export default Sentry.wrap(RootLayout)

```

And app/index.tsx

```
import { View, Text } from "react-native"

export default function HomeScreen() {
  return (
    <View>
      <Text>Hello, world.</Text>
    </View>
  )
}
```

metro.config.js

```
const path = require("path")
const { getSentryExpoConfig } = require("@sentry/react-native/metro")
const { FileStore } = require("metro-cache")

// Find the project and workspace directories
const projectRoot = __dirname

// This can be replaced with `find-pnpm-workspace-root`
const workspaceRoot = path.resolve(projectRoot, "../..")
const config = getSentryExpoConfig(__dirname)

console.log(config)

config.transformer.getTransformOptions = async () => ({
  transform: {
    experimentalImportSupport: true,
    inlineRequires: false,
  },
})

config.resolver.extraNodeModules = {
  stream: require.resolve("readable-stream"),
}

config.resolver.nodeModulesPaths = [
  path.resolve(projectRoot, "node_modules"),
  path.resolve(workspaceRoot, "node_modules"),
]

config.resolver.disableHierarchicalLookup = true

config.cacheStores = [
  new FileStore({
    root: path.join(projectRoot, "node_modules", ".cache", "metro"),
  }),
]

module.exports = config

```

### Expected Result

No error, app loads

### Actual Result

App crashes immediately.

**Comment by krystofwoldrich** [comment]
Hi @ian,
thank you for the message,

have Sentry RN package work for you with the previous Expo version?

**Comment by ian** [comment]
Yes, was working in expo 51. I’m working on getting a minimum repro pushed, standby.

**Comment by ian** [comment]
@krystofwoldrich here's a minimum reproducible example: https://github.com/ian/sentry-repro

Used https://docs.expo.dev/guides/using-sentry/ as an installation guide, and getting same issue. Let me know if you need more info.

![Image](https://github.com/user-attachments/assets/4e68c0b7-babb-483a-af61-0f89c9b74a20)

**Comment by krystofwoldrich** [comment]
Thank you.

**Comment by gunnartorfis** [comment]
Additional context, this does not happen in 6.1.0, but does in 6.3.0.

**Comment by ian** [comment]
Confirmed what @gunnartorfis suggested, downgrading to 6.1.0 fixes it.

**Comment by kahest** [comment]
thanks for confirming 👍 this should help narrow the problem down

**Comment by Jonatthu** [comment]
Confirmed what @gunnartorfis suggested, downgrading to 6.1.0 fixes it

**Comment by krystofwoldrich** [comment]
Hi @ian,
thank you for the reproducible example.

The reason for the `undefined is not a function` was a versions mismatch of `@sentry/core`. Your application depends on `@sentry/core@7.119.2` but `@sentry/react-native@6.3.0` requires `@sentry/core@8.40.0`.

https://github.com/ian/sentry-repro/blob/main/apps/mobile/package.json#L14-L15

---

Always keep the `@sentry/core` version matching with what `@sentry/react-native` uses.

**Comment by gunnartorfis** [comment]
@krystofwoldrich in my case there's no direct dependency on @sentry/core. I have a monorepo with Expo apps, Remix, Next.js and Node GraphQL and they all have Sentry. Thus when running yarn (v4) why @sentry/core, I get a bunch of versions. 

Isn't it up to @sentry/react-native to handle this somehow? I don't know how but I don't think its scalable for the projects using the lib to declare the resolution they want. 

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/855cc3d8-7299-49a7-8dc6-8d1188bd60f8" />
